### PR TITLE
Change `eltype` methods for objects to types

### PR DESCRIPTION
**Pull request by lgoettgens**
The docstring of `eltype` states
> However the form that accepts a type argument should be defined for new types. 

Chaning this should help julia infer types of iterator constructions like `Iterators.reverse(...)` for these types.

### CODE REVIEW

File: src/InvariantTheory/iterators.jl [hunk 1]
```diff
@@ -391,6 +391,7 @@ function iterate_basis_linear_algebra(IR::FinGroupInvarRing, d::Int)
 end
 
 Base.eltype(BI::FinGroupInvarRingBasisIterator) = elem_type(polynomial_ring(BI.R))
+# TODO: change this to Base.eltype(::Type{<:FinGroupInvarRingBasisIterator{FinGroupInvarRingT}}) where {FinGroupInvarRingT} = ...
```
Comment (lgoettgens) [inline src/InvariantTheory/iterators.jl]: @joschmitt do you maybe have an idea how to do this more cleanly? I am a bit overwhelmed by the amount of type vars and what to use here

Comment (joschmitt) [reply src/InvariantTheory/iterators.jl]: Should be possible, I'll have a look tomorrow.

File: test/PolyhedralGeometry/subdivision_of_points.jl [hunk 2]
```diff
@@ -14,8 +14,8 @@
     SubdivisionOfPoints
 
   @testset "alternative inputs" begin
-    @test collect(maximal_cells(square_by_incidence)) ==
-      collect(maximal_cells(square_by_weights))
+    @test issetequal(collect(maximal_cells(square_by_incidence)),
+      collect(maximal_cells(square_by_weights)))
```
Comment (lgoettgens) [inline test/PolyhedralGeometry/subdivision_of_points.jl:18]: @benlorenz just fyi I patched another one (or the same, cannot remember) of these flaky tests, as it failed here

Comment (benlorenz) [reply test/PolyhedralGeometry/subdivision_of_points.jl:18]: I think this is the same one.
I am a bit surprised why this is now happening again and again without any related changes (and only on the macos runner?). @alexej-jordan is working on fixing this and more similar tests.

File: src/InvariantTheory/iterators.jl (hunk 1 above)
Comment (joschmitt) [reply src/InvariantTheory/iterators.jl]: Should be resolved.

**Comment by fingolfin** [comment]
Code formatting CI test fails

**Comment by joschmitt** [comment]
> Code formatting CI test fails

Yeah, my fault, fixed.

**Review by fingolfin — APPROVED** [review]

File: src/InvariantTheory/iterators.jl (hunk 1 above)
Comment (lgoettgens) [reply src/InvariantTheory/iterators.jl]: Thanks,!

**Pull request merged**